packages/zoe/src/contracts/multipoolAutoswap/swap.js

```diff
@@ -0,0 +1,136 @@
+import { assertProposalKeywords, trade } from '../../contractSupport';
+
+import '../../../exported';
+
+/**
+ *
+ * @param {ContractFacet} zcf
+ * @param {(brand: Brand) => boolean} isSecondary
+ * @param {(brand: Brand) => boolean} isCentral
+ * @param {(brand: Brand) => Pool} getPool
+ */
+export const makeMakeSwapInvitation = (
+  zcf,
+  isSecondary,
+  isCentral,
+  getPool,
+) => {
+  const swapExpected = harden({
+    give: { In: null },
+    want: { Out: null },
+  });
+
+  const swap = seat => {
+    const {
+      give: { In: amountIn },
+      want: { Out: wantedAmountOut },
+    } = seat.getProposal();
+    const { brand: brandIn, value: inputValue } = amountIn;
+    const brandOut = wantedAmountOut.brand;
+
+    // we could be swapping (1) central to secondary, (2) secondary to
+    // central, or (3) secondary to secondary
```

Comment:
This was mind-numbing to read.  It would be clearer to me (but YMMV) if we set up the variables, then did:
```js
if (secondaryIn) {
  trade secondaryIn to central;
}
if (secondaryOut) {
  trade central to secondaryOut;
}
seat.exit();
```

Reply:
with seat staging this might be possible now.  I think you're right that that would be shorter adn simpler to read.